usr/src/uts/common/fs/zfs/dmu_send.c

```diff
@@ -2206,6 +2215,9 @@ receive_write(struct receive_writer_arg *rwa, struct drr_write *drrw,
 	rwa->last_object = drrw->drr_object;
 	rwa->last_offset = drrw->drr_offset;
 
+	if (rwa->last_object > rwa->max_object)
+		rwa->max_object = rwa->last_object;
```

Comment:
I think we would always see the OBJECT record before WRITE* records, even when resuming.  But I guess it's fine to check this here too.

Reply:
Given that this is a port of an existing patch, I'd rather leave it identical to the ZoL version to reduce future conflicts.